src/transport/SessionManager.h

```diff
@@ -174,8 +174,155 @@ class DLL_EXPORT SessionManager : public TransportMgrDelegate, public FabricTabl
+
+            //
+            // It's entirely possible to either come across a PASE session OR, a CASE session
+            // that has yet to be activated (i.e a CASEServer holding onto a SecureSession object
+            // waiting for a Sigma1 message to arrive). Let's skip those.
+            //
+            if (!session->IsCASESession() || session->GetFabricIndex() == kUndefinedFabricIndex)
+            {
+                return Loop::Continue;
+            }
+
+            auto * compareFabric = mFabricTable->FindFabricWithIndex(session->GetFabricIndex());
+            VerifyOrDie(compareFabric != nullptr);
+
+            err = compareFabric->FetchRootPubkey(comparePubKey);
+            VerifyOrDie(err == CHIP_NO_ERROR);
+
+            if (comparePubKey.Matches(targetPubKey) && targetFabric->GetFabricId() == compareFabric->GetFabricId() &&
+                session->GetPeerNodeId() == node.GetNodeId())
+            {
+                function(session);
+            }
+
+            return Loop::Continue;
+        });
+
+        return CHIP_NO_ERROR;
+    }
+
+    /**
+     * @brief
+     *
+     * Call the provided lambda on all sessions that match the logical fabric
+     * associated with the provided fabric index.
+     *
+     */
+    template <typename Function>
+    CHIP_ERROR ForEachCollidingSession(FabricIndex fabricIndex, Function && function)
+    {
+        Crypto::P256PublicKey targetPubKey;
+
+        auto * targetFabric = mFabricTable->FindFabricWithIndex(fabricIndex);
+        VerifyOrReturnError(targetFabric != nullptr, CHIP_ERROR_INVALID_FABRIC_INDEX);
+
+        auto err = targetFabric->FetchRootPubkey(targetPubKey);
+        VerifyOrDie(err == CHIP_NO_ERROR);
+
+        mSecureSessions.ForEachSession([&](auto * session) {
+            Crypto::P256PublicKey comparePubKey;
+
+            //
+            // It's entirely possible to either come across a PASE session OR, a CASE session
+            // that has yet to be activated (i.e a CASEServer holding onto a SecureSession object
+            // waiting for a Sigma1 message to arrive). Let's skip those.
+            //
+            if (!session->IsCASESession() || session->GetFabricIndex() == kUndefinedFabricIndex)
+            {
+                return Loop::Continue;
+            }
+
+            auto * compareFabric = mFabricTable->FindFabricWithIndex(session->GetFabricIndex());
+            VerifyOrDie(compareFabric != nullptr);
+
+            err = compareFabric->FetchRootPubkey(comparePubKey);
+            VerifyOrDie(err == CHIP_NO_ERROR);
+
+            if (comparePubKey.Matches(targetPubKey) && targetFabric->GetFabricId() == compareFabric->GetFabricId())
+            {
+                function(session);
+            }
+
+            return Loop::Continue;
+        });
+
+        return CHIP_NO_ERROR;
+    }
+
     void ExpireAllSessions(const ScopedNodeId & node);
     void ExpireAllSessionsForFabric(FabricIndex fabricIndex);
```

Comment:
This method no longer does what its name implies.  This could sort of work:

```suggestion
    void ExpireAllSessionsForFabricIndex(FabricIndex fabricIndex);
```

However, in the spec, fabricIndex is a well-defined thing that unambiguously maps to a single fabric.  So this is confusion inducing.

I understand why what we're doing here is functionally OK: the 1:1 mapping between fabric index and fabric is only necessary to fulfill the outward data contract when serving the Operational Credentials Cluster.  Still, this is confusing.

If we have an API that purports to `ExpireAllSessionsForFabric`, I think it should actually do that; i.e., there should be internal magic that also expires sessions for other fabric indices with a matching fabric identity.

If we are not going to do that, I think the new APIs shouldn't be compiled into the code by default, because they cause the preexisting APIs to no longer fulfill their full function.